quinn-proto/src/endpoint.rs

```diff
@@ -238,54 +238,18 @@ impl Endpoint {
         //
         // Potentially create a new connection
         //
-
         let dst_cid = first_decode.dst_cid();
-        let Some(server_config) = &self.server_config else {
```

Comment:
If we're a client receiving a short-header packet, this change will cause us to reach the end of the function rather than exiting early. That should be fine, since the behavior is to send a stateless reset regardless, but the comment "If we got this far, we're a server ..." must be updated.